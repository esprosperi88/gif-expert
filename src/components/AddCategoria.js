
import React, { useState } from 'react';
import PropsTypes from 'prop-types';

export const AddCategoria =( {setCategorias}) =>{

    const [inputValue,setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };
    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.length>2){
            setCategorias(cat =>[ inputValue,...cat ]);
            setInputValue('');
        }

        console.log(inputValue);
    };

    return (<form onSubmit={handleSubmit}>
                <p>{inputValue}</p>
                <input 
                    type="text"
                    value={inputValue}
                    onChange={handleInputChange}
                /> 
            </form>
    );

}

AddCategoria.propTypes={

    setCategorias: PropsTypes.func.isRequired

}


export default AddCategoria;