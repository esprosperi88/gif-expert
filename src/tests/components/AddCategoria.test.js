import React from "react";
import AddCategoria from '../../components/AddCategoria';
import {shallow}   from 'enzyme'


describe('Prueba component - AddCategoria ',() => {

    const setCategorias = () => {};

    let wrapper;

    beforeEach( () => {        
        wrapper = shallow(<AddCategoria 
            setCategorias={setCategorias} />);
    });



    test('AddCategoria - snapshot',() => {
        expect (wrapper).toMatchSnapshot();
    });

    test('AddCategoria - debe de cambiar la caja de texto',() => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change',{target:{value}});
        wrapper.find('p').text()
        expect(wrapper.find('p').text().trim()).toBe(value);
        
        
        
    });




});