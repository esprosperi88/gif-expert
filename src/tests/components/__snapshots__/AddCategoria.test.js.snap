// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba component - AddCategoria  AddCategoria - snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
